### PR TITLE
Set up initial project skeleton for managed Blox

### PR DESCRIPTION
**Pull request by wjbuys**
#### Summary
This sets up an empty multi-project Gradle build for Blox (see #211 for details).

#### Implementation details
This is what the project looks like:
```
------------------------------------------------------------
Root project - Blox: Open Source schedulers for Amazon ECS
------------------------------------------------------------

Root project 'blox' - Blox: Open Source schedulers for Amazon ECS
+--- Project ':buildSrc' - Custom build logic for building Blox
+--- Project ':data-model' - Shared POJOs and interfaces for the Blox DataService
+--- Project ':data-service' - Handler implementations for the Blox DataService
+--- Project ':frontend' - Lambda-backed API Gateway API for the Blox Frontend API
+--- Project ':frontend-client' - Auto-generated client for the Blox Frontend API
\--- Project ':frontend-infrastructure' - AWS Cloudformation templates and deployment scripts for the Blox Frontend API
```
#### Testing
I've added a `.travis.yml` that will run `gradle check` in the build, which should succeed (because there's nothing in here yet):

```
$ gradle check

BUILD SUCCESSFUL in 1s
```

#### Licensing
This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

**Comment by wjbuys** [comment]
Hm, should we also copy over the design docs from the current `dev` branch?

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,3 @@
+group 'com.amazonaws.blox'
+version '0.0-SNAPSHOT'
```
Comment (narehayrapetyan) [inline build.gradle]: why 0.0?

File: data-model/build.gradle [hunk 2]
```diff
@@ -0,0 +1,3 @@
+description 'Shared POJOs and interfaces for the Blox DataService'
```
Comment (narehayrapetyan) [inline data-model/build.gradle]: data-service-model? it's not the data model which is confusing

File: data-service/build.gradle [hunk 3]
```diff
@@ -0,0 +1,3 @@
+description "Handler implementations for the Blox DataService"
```
Comment (narehayrapetyan) [inline data-service/build.gradle]: I think handler is confusing. Data service implementation covers it imo

**Review by narehayrapetyan — CHANGES_REQUESTED** [review]
I know we talked about pushing this to a different branch but since I moved everything to v0.3 branch already and there's no way to rename branches I think we should just rebase this on dev and push directly to dev. We also won't need to copy things like docs over.

File: build.gradle (hunk 1 above)
Comment (wjbuys) [reply build.gradle]: It's totally non-working at the moment. Should I make it 0.1 instead?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+language: java
+before_cache:
```
Comment (narehayrapetyan) [inline .travis.yml]: don't we need to have a build step here?

File: data-model/build.gradle (hunk 2 above)
Comment (wjbuys) [reply data-model/build.gradle]: Do you have a suggestion? What do we call the packages that contain the DAO POJOs and Java interfaces that define each Lambda microservice?

File: data-service/build.gradle (hunk 3 above)
Comment (wjbuys) [reply data-service/build.gradle]: Yeah, handler is vague. I was thinking of the entrypoint interface that all Lambda functions implement, i.e. `RequestHandler`. Will change to just "Implementation of the Blox DataService"

File: .travis.yml (hunk 4 above)
Comment (wjbuys) [reply .travis.yml]: The docs linked in the commit message explain how this works, I'll add a comment inline too.

Basically, travis configures java/gradle builds automatically if the repo contains a `build.gradle` (see https://docs.travis-ci.com/user/languages/java/#Projects-Using-Gradle)

File: data-model/build.gradle (hunk 2 above)
Comment (narehayrapetyan) [reply data-model/build.gradle]: I was suggesting dataservice-model

File: build.gradle (hunk 1 above)
Comment (narehayrapetyan) [reply build.gradle]: Sounds good

Comment (wjbuys) [reply build.gradle]: Okay, fixed it now

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+### Before you start
+If you would like to make a significant change, it's a good idea to first open an issue to discuss it.
+
+### Making the request
+Development takes place against the dev branch of this repository and pull requests should be opened against that branch.
+
+### Testing
+Any contributions should pass all tests.
+You may run all tests by running `make` inside cluster-state-service and daemon-scheduler (requires go, go-swagger to be installed). If you have golang installed, you may run `make get-deps` to install dependencies.
```
Comment (narehayrapetyan) [inline CONTRIBUTING.md:12]: this needs to be removed.

**Review by narehayrapetyan — COMMENTED** [review]
Other than removing the go makefile comment in the contributing doc, looks good.

**Review by narehayrapetyan — APPROVED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (wjbuys) [reply CONTRIBUTING.md:12]: Gotcha, will do

**Comment by wjbuys** [comment]
This has been force-pushed into `blox:dev`.

Edit: Just to explain the force push here, we were faced with two options:
- continue this as a continuous line of history from the current `dev` branch. This would introduce a commit with a huge diff of "delete everything", and would force us to carry around the history of the unrelated GoLang codebase in every (non-shallow) clone. Users pulling the dev branch would immediately get conflicts for all their changes (since we deleted all files).
- create this as a new orphan branch and start from scratch, then force push into `dev` to switch over the history. New clones get the actually-in-development code, with none of the unrelated baggage. The old code is still available in the `v0.3` branch. If anyone has cloned the repo at the `dev` branch and tries to pull, they'll get slightly more cryptic conflicts for all changes.

Neither of these are great options; we opted to go for keeping the history separate, and dealing with the fallout of users' clones of the dev branch no longer being pullable. The solution in either case is for users to switch to the `v0.3` branch in their working copy, and pull from that.

**Pull request merged**